modules/ossm-federation-config-meshPeer.adoc

```diff
@@ -0,0 +1,111 @@
+* service_mesh/v2x/ossm-federation.adoc
+////
+
+[id="ossm-task_{context}"]
+= Joining a federated mesh
+
+You declare the federation between two meshes by creating a `ServiceMeshPeer` resource. This resource defines the federation between two meshes, and you use it to configure discovery for the peer mesh, access to the peer mesh, and certificates used to validate the other mesh’s clients.
+
+//Insert ServiceMeshPeer diagram here
+
+Configuration in the `ServiceMeshPeer` resource includes the following:
+
+* The address of the other mesh’s ingress gateway, which is used for discovery and service requests.
+* The names of the local ingress and egress gateways that is used for interactions with the specified peer mesh.
+* The client ID used by the other mesh when sending requests to this mesh.
+* The trust domain used by the other mesh.
+* The name of a `ConfigMap` containing a root certificate that is used to validate client certificates in the trust domain used by the other mesh.
+
+Meshes are federated on a one-to-one basis, so each pair of peers requires a `ServiceMeshPeer` file specifying the federation connection to the other service mesh. For example, federating three meshes named `red`, `blue`, and `green` would require six `ServiceMeshPeer` files.
+
+. On red-mesh-system, create a `ServiceMeshPeer` for the green mesh.
+. On red-mesh-system, create a `ServiceMeshPeer` for the blue mesh.
+. On green-mesh-system, create a `ServiceMeshPeer` for the red mesh.
+. On green-mesh-system, create a `ServiceMeshPeer` for the blue mesh.
+. On blue-mesh-system, create a `ServiceMeshPeer` for the red mesh.
+. On blue-mesh-system, create a `ServiceMeshPeer` for the green mesh.
+
+In the following example, the administrator for the `red-mesh` is configuring federation with the `green-mesh`.
+
+.Example ServiceMeshPeer resource for red-mesh
+[source,yaml]
+----
+kind: ServiceMeshPeer
+apiVersion: federation.maistra.io/v1
+metadata:
+  name: green-mesh
+  namespace: red-mesh-system
+spec:
+  remote:
+    addresses:
+    - ingress-red-mesh.red-mesh-system.apps.domain.com
+  gateways:
+    ingress:
+      name: ingress-green-mesh
+    egress:
+      name: egress-green-mesh
+  security:
+    trust:
+      domain: green-mesh.local
+    clientID: green-mesh.local/ns/green-mesh-system/sa/egress-red-mesh-service-account
+    certificateChain:
+      kind: ConfigMap
+      name: green-mesh-ca-root-cert
+----
+
+.ServiceMeshPeer configuration parameters
+[options="header"]
+[cols="l, a, a"]
+|===
+|Parameter |Description |Values
+|metadata:
+  name:
+|Name of the peer mesh that this file is configuring federation with.
+|String
+
+|metadata:
+  namespace:
+|System namespace for this mesh, that is, where the mesh control plane is installed.
+|String
+
+|spec:
+  remote:
+    addresses:
+|List of addresses of the peer meshes' ingress gateways that are servicing requests from this mesh.
+|<IngressName>.<SystemName>.apps.domain.com
+
```

Comment:
Missing two entries:
- `spec.remote.discoveryPort` (the port on which the addresses are handling discovery requests; defaults to 8188)
- `spec.remote.servicePort` (the port on which the addresses are handling service requests; defaults to 15443)